flang/lib/Lower/IO.cpp

```diff
@@ -354,14 +355,6 @@ static void genInputItemList(Fortran::lower::AbstractConverter &converter,
     }
     ok = builder.create<mlir::CallOp>(loc, inputFunc, inputFuncArgs)
              .getResult(0);
-    if (complexPartType) { // imaginary part
```

Comment:
Is the runtime now assuming a complex is going to be passed as a pair of {float|double} in memory as `(real, imaginary)`?

Reply:
Yes, a pointer to such a structure.